test/e2e/specs/editor/various/list-view.spec.js

```diff
@@ -0,0 +1,345 @@
+};
+
+test.describe( 'List View', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'allows a user to drag a block to a new sibling position', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Ensure the setup is correct before dragging.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [
+				{ name: 'core/heading' },
+				{ name: 'core/image' },
+				{ name: 'core/paragraph' },
+			] );
+
+		// Drag the paragraph above the heading.
+		const paragraphBlockItem = page.getByRole( 'gridcell', {
+			name: 'Paragraph link',
+		} );
+		const headingBlockItem = page.getByRole( 'gridcell', {
+			name: 'Heading link',
+		} );
+		await paragraphBlockItem.dragTo( headingBlockItem, { x: 0, y: 0 } );
+
+		// Ensure the block was dropped correctly.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [
+				{ name: 'core/paragraph' },
+				{ name: 'core/heading' },
+				{ name: 'core/image' },
+			] );
+	} );
+
+	// Check for regressions of https://github.com/WordPress/gutenberg/issues/38763.
+	test( 'shows the correct amount of blocks after a block is removed in the canvas', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Go to the image block in List View.
+		await pageUtils.pressKeyTimes( 'ArrowUp', 2 );
+		await expect(
+			page
+				.getByRole( 'gridcell', {
+					name: 'Image link',
+				} )
+				.locator( 'a' )
+		).toBeFocused();
+
+		// Select the image block in the canvas.
+		await page.keyboard.press( 'Enter' );
+		const imageBlock = page.getByRole( 'document', {
+			name: 'Block: Image',
+		} );
+		await expect(
+			imageBlock.getByRole( 'button', { name: 'Upload' } )
+		).toBeFocused();
+		await page.keyboard.press( 'ArrowUp' );
+		await expect( imageBlock ).toBeFocused();
+
+		// Start listeninig to console errors.
+		let hasThrownConsoleError = false;
+		page.on( 'console', ( msg ) => {
+			if ( msg.type() === 'error' ) {
+				hasThrownConsoleError = true;
+			}
+		} );
+
+		// Delete the image block in the canvas.
+		await page.keyboard.press( 'Backspace' );
+
+		// List View should have two rows.
+		await expect(
+			page.getByRole( 'gridcell', { name: /link/i } )
+		).toHaveCount( 2 );
+
+		// Ensure console didn't throw an error as reported in
+		// https://github.com/WordPress/gutenberg/issues/38763.
+		await expect( hasThrownConsoleError ).toBeFalsy();
```

Comment:
I don't think `toBeFalsy()` returns a promise?

```suggestion
		expect( hasThrownConsoleError ).toBeFalsy();
```

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/47919/commits/5aa48826c3d979b2d1abc0e17e55ab5650fd94b3 (also, see https://github.com/WordPress/gutenberg/pull/47919#discussion_r1103790059)